sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/Secret.java

```diff
@@ -52,76 +62,78 @@ public String getValue() {
     }
 
     /**
-     * Set enabled value.
+     * Get the secret identifier.
      *
-     * @param enabled The enabled value to set
-     * @return the Secret object itself.
+     * @return the secret identifier.
      */
-    @Override
-    public Secret setEnabled(Boolean enabled) {
-        super.setEnabled(enabled);
+    public Secret getId() {
+        properties.getId();
         return this;
     }
 
     /**
-     * Set the {@link OffsetDateTime notBefore} UTC time.
+     * Get the secret name.
      *
-     * @param notBefore The not before time value to set
-     * @return the Secret object itself.
+     * @return the secret name.
      */
-    @Override
-    public Secret setNotBefore(OffsetDateTime notBefore) {
-        super.setNotBefore(notBefore);
-        return this;
+    public String getName() {
+        return properties.getName();
     }
 
     /**
-     * Set the {@link OffsetDateTime expires} UTC time.
-     *
-     * @param expires The expiry time value to set
-     * @return the Secret object itself.
+     * Get the secret properties
+     * @return the Secret properties
      */
-    @Override
-    public Secret setExpires(OffsetDateTime expires) {
-        super.setExpires(expires);
-        return this;
+    public SecretProperties getProperties() {
+        return this.properties;
     }
 
     /**
-     * Set the secret identifier.
-     *
-     * @param id The secret identifier value to set
-     * @return the Secret object itself.
+     * Set the secret properties
+     * @param properties The Secret properties
+     * @return the updated secret object
      */
-    @Override
-    public Secret setId(String id) {
-        super.setId(id);
+    public Secret setProperties(SecretProperties properties) {
+        properties.name = this.properties.name;
+        this.properties = properties;
         return this;
```

Comment:
Should properties be setable? Also what if properties argument is null?

Reply:
added the null check.
setters are idiomatic in java, facilitates fluency.

Reply:
as we discussed.